akka-http2-support/src/main/scala/akka/http/scaladsl/Http2.scala

```diff
@@ -103,33 +105,65 @@ final class Http2Ext(private val config: Config)(implicit val system: ActorSyste
     settings:    ServerSettings,
     parallelism: Int,
     log:         LoggingAdapter
-  ): HttpRequest ⇒ Future[HttpResponse] = { req ⇒
+  )(implicit mat: Materializer): HttpRequest ⇒ Future[HttpResponse] = { req ⇒
     req.header[Upgrade] match {
       case Some(upgrade) if upgrade.protocols.exists(_.name equalsIgnoreCase "h2c") ⇒
-        // TODO remove duplication?
-        val serverLayer: Flow[ByteString, ByteString, Future[Done]] = Flow.fromGraph(
-          Flow[HttpRequest]
-            .watchTermination()(Keep.right)
-            // FIXME: parallelism should maybe kept in track with SETTINGS_MAX_CONCURRENT_STREAMS so that we don't need
-            // to buffer requests that cannot be handled in parallel
-            .via(Http2Blueprint.handleWithStreamIdHeader(parallelism)(handler)(system.dispatcher))
-            .joinMat(Http2Blueprint.serverStack(settings, log))(Keep.left))
-
-        // TODO consume the HTTP2-Settings header
-        // TODO consume the request entity and make sure it is inserted as an HTTP2 request
-        // https://http2.github.io/http2-spec/#rfc.section.3.2
-
-        Future.successful(
-          HttpResponse(
-            StatusCodes.SwitchingProtocols,
-            immutable.Seq[HttpHeader](
-              ConnectionUpgradeHeader,
-              UpgradeHeader,
-              // TODO add attributes from the request
-              UpgradeToOtherProtocolHeader(serverLayer)
+
+        log.debug("Got h2c upgrade request from HTTP/1.1 to HTTP2")
+
+        // https://http2.github.io/http2-spec/#Http2SettingsHeader 3.2.1 HTTP2-Settings Header Field
+        val upgradeSettings = req.headers.collect {
```

Comment:
As the spec promises exactly 1 header here, `collect` can become `find` + `map`?

Reply:
Ah, you quote 'A server MUST NOT upgrade the connection to HTTP/2 if this header field is not present or if more than one is present' below, :+1: then!